agent/consul/internal_endpoint.go

```diff
@@ -130,23 +145,36 @@ func (m *Internal) KeyringOperation(
 	return nil
 }
 
-// executeKeyringOp executes the appropriate keyring-related function based on
-// the type of keyring operation in the request. It takes the KeyManager as an
-// argument, so it can handle any operation for either LAN or WAN pools.
+// executeKeyringOp executes the keyring-related operation in the request
+// on either the WAN or LAN pools.
 func (m *Internal) executeKeyringOp(
 	args *structs.KeyringRequest,
 	reply *structs.KeyringResponses,
 	wan bool) {
 
-	var serfResp *serf.KeyResponse
-	var err error
-	var mgr *serf.KeyManager
-
 	if wan {
-		mgr = m.srv.KeyManagerWAN()
+		mgr := m.srv.KeyManagerWAN()
+		m.executeKeyringOpMgr(mgr, args, reply, wan)
 	} else {
-		mgr = m.srv.KeyManagerLAN()
+		segments := m.srv.LANSegments()
```

Comment:
Doh this doesn't special case for the default segment, see comment above we should probably just put the default segment in here and avoid the special cases.